src/utils/ipv6.hpp

```diff
@@ -0,0 +1,45 @@
+#pragma once
+
+#include <cstring>
+#include <iterator>
+#include <arpa/inet.h>
+
+#include "ipv6.hpp"
+
+namespace
+{
+    constexpr char ipv4_mapped_ipv6_prefix[] = "::ffff:";
+    constexpr std::size_t ipv4_mapped_ipv6_prefix_length =
+        std::size(ipv4_mapped_ipv6_prefix);
```

Comment:
Oh I see :+1: 

With the previous change "constexpr auto IPV4_MAPPED_IPV6_PREFIX" = "::ffff:"sv;" , I think it will be fixed by using the IPV4_MAPPED_IPV6_PREFIX.length() directly which count char number until meet "\0" at compile-time